src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/jpa/UserPersonDetail.kt

```diff
@@ -1,37 +1,66 @@
 package uk.gov.justice.digital.hmpps.nomisuserrolesapi.jpa
 
 import org.hibernate.Hibernate
+import org.hibernate.annotations.Where
+import javax.persistence.CascadeType
 import javax.persistence.Column
 import javax.persistence.Entity
 import javax.persistence.FetchType
 import javax.persistence.Id
 import javax.persistence.JoinColumn
+import javax.persistence.JoinTable
+import javax.persistence.ManyToMany
 import javax.persistence.ManyToOne
 import javax.persistence.OneToMany
 import javax.persistence.Table
 
+@Suppress("DataClassEqualsAndHashCodeInspection")
 @Entity
 @Table(name = "STAFF_USER_ACCOUNTS")
 data class UserPersonDetail(
   @Id
   @Column(name = "USERNAME", nullable = false)
   val username: String,
 
-  @ManyToOne
+  @ManyToOne(cascade = [CascadeType.ALL])
   @JoinColumn(name = "STAFF_ID")
   val staff: Staff,
 
   @OneToMany(fetch = FetchType.LAZY)
   @JoinColumn(name = "USERNAME")
-  val roles: List<UserCaseloadRole> = listOf()
-) {
+  val roles: List<UserCaseloadRole> = listOf(),
+
+  @ManyToMany(fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
+  @JoinTable(name = "USER_ACCESSIBLE_CASELOADS", joinColumns = [JoinColumn(name = "USERNAME", referencedColumnName = "USERNAME")], inverseJoinColumns = [JoinColumn(name = "CASELOAD_ID", referencedColumnName = "CASELOAD_ID")])
+  val caseloads: List<Caseload> = listOf(),
+
```

Comment:
will change this so we have the "accessible caseload" entity so the START_DATE on the table can be set (even though it is nullable NOMIS might require it)